functional-tests/fn_bridge_deposit_happy.py

```diff
@@ -0,0 +1,157 @@
+import time
+from typing import List
+
+import flexitest
+from bitcoinlib.services.bitcoind import BitcoindClient
+
+from constants import DEFAULT_ROLLUP_PARAMS
+from entry import BasicEnvConfig
+from utils import get_logger
+
+
+@flexitest.register
+class BridgeDepositHappyTest(flexitest.Test):
+    """
+    A test class for happy path scenarios of bridge deposits.
+
+    It tests the functionality of depositing Bitcoin into the bridge and verifying the
+    corresponding increase in balance on the Ethereum side.
+
+    Deposit funds into the rollup, happy path,
+    check that funds can be swept with a normal Ethereum transaction afterwards.
+    """
+
+    def __init__(self, ctx: flexitest.InitContext):
+        ctx.set_env(BasicEnvConfig(101))
+        self.logger = get_logger("BridgeDepositHappyTest")
+
+    def main(self, ctx: flexitest.RunContext):
+        btc = ctx.get_service("bitcoin")
+        seq = ctx.get_service("sequencer")
+        el_address_1 = "deadf001900dca3ebeefdeadf001900dca3ebeef"
+        el_address_2 = "deedf001900dca3ebeefdeadf001900dca3ebeef"
+
+        seqrpc = seq.create_rpc()
+        btcrpc: BitcoindClient = btc.create_rpc()
+        reth = ctx.get_service("reth")
+        rethrpc = reth.create_rpc()
+
+        original_balance = int(rethrpc.eth_getBalance(f"0x{el_address_1}"), 16)
+        self.logger.debug(f"Balance before deposit: {original_balance}")
+
+        original_balance_2 = int(rethrpc.eth_getBalance(f"0x{el_address_2}"), 16)
+        self.logger.debug(f"Balance before deposit: {original_balance_2}")
+
+        original_num_deposits = len(seqrpc.strata_getCurrentDeposits())
+        self.logger.debug(f"Original deposit count: {original_num_deposits}")
+
+        fees_in_btc = 0.01
+        sats_per_btc = 10**8
+        amount_to_send = DEFAULT_ROLLUP_PARAMS["deposit_amount"] / sats_per_btc + fees_in_btc
+        take_back_leaf_hash = "02" * 32
+        magic_bytes = DEFAULT_ROLLUP_PARAMS["rollup_name"].encode("utf-8").hex()
+        options = {"changePosition": 2}
+
+        current_block_num = int(rethrpc.eth_blockNumber(), base=16)
+        print(f"Current reth block num: {current_block_num}")
+
+        # First deposit address 1
+        addr_1 = btcrpc.proxy.getnewaddress("", "bech32m")
+        btcrpc.proxy.generatetoaddress(1, addr_1)
+        outputs = [
+            {addr_1: amount_to_send},
+            {"data": f"{magic_bytes}{take_back_leaf_hash}{el_address_1}"},
+        ]
```

Comment:
This is a Deposit Transaction Request (DRT), right?
Hence, this should have the following outputs:

1. `OP_RETURN` specified in the second ordered list in [Strata docs bridge](https://docs.stratabtc.org/technical/bitcoin-bridge/#deposit)
2. a P2TR address that cannot be spent from a keypath spend and has a two spending paths specified in the fist ordered list in [Strata docs bridge](https://docs.stratabtc.org/technical/bitcoin-bridge/#deposit)

If so, I'll probably create a bunch of helper functions in `functional_tests/utils.py` that compute these addresses and `OP_RETURN`s, what you think?

Reply:
This is a good approach. :+1: 

I had in mind that we could just directly invoke the CLI tool as a subproc, but perhaps instead we should duplicate some of these tests so that we can separate test deposits manually constructed from Python from tests using the CLI tool.  So if the CLI tool breaks but this test doesn't (or vice versa) we get more information about where the trouble lies.